port_dsa.c

```diff
@@ -0,0 +1,83 @@
+// The following is a partial backport of crypto/dsa/dsa_pmeth.c,
+// commit cbc8a839959418d8a2c2e3ec6bdf394852c9501e on the
+// OpenSSL_1_1_0-stable branch.  The ctrl function has been removed.
```

Comment:
Why was the ctrl function removed?  Not needed for our use case?

I assume the license of the openssl code is compatible for us to include this backport in this repo?

Reply:
> Why was the ctrl function removed? Not needed for our use case?

This comment is outdated. I've updated it with something more meaningful.

> I assume the license of the openssl code is compatible for us to include this backport in this repo?

Not sure about this. This backporting approach is taken from the [.NET Runtime](https://github.com/dotnet/runtime/blob/main/src/native/libs/System.Security.Cryptography.Native/openssl_1_0_structs.h), and its third party notices doesn't mention OpenSSL, so I guess we are fine. My take is that these are just some getters and setters adapted for our need, nothing really interesting that worth mentioning, but I could be wrong. Let's talk about this in our next meeting.